apiserver/client/status.go

```diff
@@ -533,21 +635,50 @@ func makeMachineStatus(machine *state.Machine) (status params.MachineStatus) {
 		}
 		status.DNSName = addr.Value
 
-		mAddrs := machine.Addresses()
-		if len(mAddrs) == 0 {
-			logger.Debugf("no IP addresses fetched for machine %q", instid)
-			// At least give it the newly created DNSName address, if it exists.
-			if addr.Value != "" {
-				mAddrs = append(mAddrs, addr)
+		status.NetworkInterfaces = make(map[string]params.NetworkInterface, len(linkLayerDevices))
+		for _, llDev := range linkLayerDevices {
+			device := llDev.Name()
+			ips := []string{}
+			gw := []string{}
+			ns := []string{}
+			sp := make(set.Strings)
+			for _, ipAddress := range ipAddresses {
+				if ipAddress.DeviceName() == device {
+					ips = append(ips, ipAddress.Value())
+					// We don't expect to find more than
+					// one ipAddress on a device with a
+					// list of nameservers, but append in
+					// any case.
+					if len(ipAddress.DNSServers()) > 0 {
```

Comment:
len check is redundant, you can just call append and it'll be happy to expand an empty slice